File: output/ChatGPT/codenet/C++/Java/s620821061.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

class BitMatching<V> {
    private List<List<Integer>> G;
    private int[] match;
    private boolean[] used;

    BitMatching() {
        this.G = new ArrayList<>();
        this.match = new int[2 * V];
        this.used = new boolean[2 * V];
        for(int i = 0; i < 2 * V; i++) {
            G.add(new ArrayList<>());
        }
    }

    void add(int a, int b) {
        G.get(a).add(b + V);
        G.get(b + V).add(a);
    }

    boolean dfs(int v) {
        used[v] = true;
        for (int i = 0; i < G.get(v).size(); i++) {
            int u = G.get(v).get(i);
            int w = match[u];
            if (w < 0 || (!used[w] && dfs(w))) {
                match[v] = u;
                match[u] = v;
                return true;
            }
        }
        return false;
    }

    int exec() {
        int res = 0;
        Arrays.fill(match, -1);
        for (int v = 0; v < V; v++) {
            if (match[v] < 0) {
                Arrays.fill(used, false);
                if (dfs(v)) {
                    res++;
                }
            }
        }
        return res;
    }
}

public class s620821061 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int n = sc.nextInt();
            int m = sc.nextInt();
            if (n == 0) break;
            BitMatching<Integer> bm = new BitMatching<>();
            int[] a = new int[n];
            int[] b = new int[m];
            for (int i = 0; i < n; i++) {
                a[i] = sc.nextInt();
            }
            for (int i = 0; i < m; i++) {
                b[i] = sc.nextInt();
            }

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < m; j++) {
                    if (gcd(a[i], b[j]) != 1) {
                        bm.add(i, j);
                    }
                }
            }
            System.out.println(bm.exec());
        }
    }

    private static int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
}

File: output/ChatGPT/codenet/C++/Python/s620821061.py
class BitMatching:
    def __init__(self, V):
        self.G = [[] for _ in range(2*V)]
        self.match = [-1] * (2*V)
        self.used = [False] * (2*V)

    def add(self, a, b):
        self.G[a].append(b+V)
        self.G[b+V].append(a)

    def dfs(self, v):
        self.used[v] = True
        for u in self.G[v]:
            w = self.match[u]
            if w < 0 or (not self.used[w] and self.dfs(w)):
                self.match[v] = u
                self.match[u] = v
                return True
        return False

    def exec(self):
        res = 0
        for v in range(V):
            if self.match[v] < 0:
                self.used = [False] * (2*V)
                if self.dfs(v):
                    res += 1
        return res


def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)


while True:
    n, m = map(int, input().split())
    if n == 0:
        break

    bm = BitMatching(n)
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    for i in range(n):
        for j in range(m):
            if gcd(a[i], b[j]) != 1:
                bm.add(i, j)

    print(bm.exec())